src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.cs

```diff
@@ -51,133 +56,1059 @@ internal struct TypeUnion
+
+            public void RecordCopy([In] IntPtr pvExisting, [Out] IntPtr pvNew)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetGuid(out Guid pguid)
+            {
+                pguid = Guid;
+            }
+
+            public void GetName([MarshalAs(UnmanagedType.BStr)] out string pbstrName)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetSize(out uint pcbSize)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetTypeInfo([MarshalAs(UnmanagedType.CustomMarshaler, MarshalType = "System.Runtime.InteropServices.CustomMarshalers.TypeToTypeInfoMarshaler")] out Type ppTypeInfo)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetField([In] IntPtr pvData, [In, MarshalAs(UnmanagedType.LPWStr)] string szFieldName, [MarshalAs(UnmanagedType.Struct)] out object pvarField)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetFieldNoCopy([In] IntPtr pvData, [In, MarshalAs(UnmanagedType.LPWStr)] string szFieldName, [MarshalAs(UnmanagedType.Struct)] out object pvarField, out IntPtr ppvDataCArray)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void PutField([In] uint wFlags, [In, Out] IntPtr pvData, [In, MarshalAs(UnmanagedType.LPWStr)] string szFieldName, [In, MarshalAs(UnmanagedType.Struct)] ref object pvarField)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void PutFieldNoCopy([In] uint wFlags, [In, Out] IntPtr pvData, [In, MarshalAs(UnmanagedType.LPWStr)] string szFieldName, [In, MarshalAs(UnmanagedType.Struct)] ref object pvarField)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void GetFieldNames([In, Out] ref uint pcNames, [MarshalAs(UnmanagedType.BStr)] out string rgBstrNames)
+            {
+                throw new NotImplementedException();
+            }
+
+            public int IsMatchingType([In, MarshalAs(UnmanagedType.Interface)] IRecordInfo pRecordInfo)
+            {
+                throw new NotImplementedException();
+            }
+
+            public IntPtr RecordCreate()
+            {
+                throw new NotImplementedException();
+            }
+
+            public void RecordCreateCopy([In] IntPtr pvSource, out IntPtr ppvDest)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void RecordDestroy([In] IntPtr pvRecord)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [Guid("0000002F-0000-0000-C000-000000000046"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+        [ComImport]
+        public interface IRecordInfo
```

Comment:
This is something we need to chat about. The entire `ITypeLib` interface family (`IRecordInfo` included) is in a weird state of support in CoreCLR. Can you please add a test ensures we fail on non-Windows platforms if APIs that use this type are called? Thanks.

cc @luqunl

Reply:
Sorry i'm slightly confused by what you mean. We have the following test already as non-Windows APIs do not implement `GetObjectForNativeVariant` (even though they probably could as this doesn't have any COM dependencies and the structure of ole variants is well documented, but the use case is minimal)

```cs
[Fact]
[PlatformSpecific(TestPlatforms.AnyUnix)]
public void GetObjectForNativeVariant_Unix_ThrowsPlatformNotSupportedException()
{
    Assert.Throws<PlatformNotSupportedException>(() => Marshal.GetObjectForNativeVariant(IntPtr.Zero));
    Assert.Throws<PlatformNotSupportedException>(() => Marshal.GetObjectForNativeVariant<int>(IntPtr.Zero));
}
```

I don't see any other examples of `IRecordInfo` being used or any APIs that take `ITypeLib` being exposed in .NET Core

Reply:
Hmm, looks like the tests don't pass against .NET Core:
```
Unhandled Exception of Type System.ArgumentException
Message :
System.ArgumentException : The specified record cannot be mapped to a managed value class.
Stack Trace :
   at System.Runtime.InteropServices.Marshal.GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
   at System.Runtime.InteropServices.Tests.GetObjectForNativeVariantTests.GetObjectForNativeVariant(Variant variant) in D:\j\workspace\windows-TGrou---74aa877a\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServices\Marshal\GetObjectForNativeVariantTests.cs:line 994
   at System.Runtime.InteropServices.Tests.GetObjectForNativeVariantTests.GetObjectForNativeVariant_Record_ReturnsExpected() in D:\j\workspace\windows-TGrou---74aa877a\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServices\Marshal\GetObjectForNativeVariantTests.cs:line 699
```

Reply:
@hughbe It absolutely has COM dependencies since the `ITypeLib` concept is based entirely around Automation/OLE - which is built on top of the COM system. My general concern was defining this interface which is an Automation interface that requires `ITypeLib` support because that is how the record is generated - from a TLB.

Reply:
Understood - sorry if i'm misunderstanding but is there anything I should action on this?

Reply:
No, you are fine. This is all good stuff. I was more concern about testing for things that we just don't want to bring in at this point. As you pointed out, there is a test for non supported platforms so at this point I am good if you are?

Reply:
yeah there is one for non supported platforms to make sure this throws PNSE